src/lists/list-item.jsx

```diff
@@ -224,8 +224,7 @@ let ListItem = React.createClass({
     React.Children.forEach(this.props.children, (child) => {
       if (child === null) return;
 
-      //consider all element to be nested except literals.
-      if (React.isValidElement(child)) {
+      if (React.isValidElement(child) && child.type.displayName === 'ListItem') {
```

Comment:
would that works to give them as primaryText?
(maybe the prop name is misleading now...)

My usecase is to have FooItem that render ListItem that are used as subitems.

Maybe we need something like:
<ListItem>
  <Whatever child>
  <NestedItem>
    whatever nested items
  </NestedItem>
</ListItem>